transport_service_queries/src/main/java/pl/edu/pg/transport/GetFlightsEventListener.java

```diff
@@ -0,0 +1,46 @@
+package pl.edu.pg.transport;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.amqp.rabbit.annotation.RabbitListener;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+@Component
+public class GetFlightsEventListener {
+    private final static Logger logger = LoggerFactory.getLogger(GetFlightsEventListener.class);
+
+    private final TransportRepository repository;
+    private final RabbitTemplate rabbitTemplate;
+    private final ObjectMapper objectMapper;
+
+    @Autowired
+    public GetFlightsEventListener(TransportRepository repository, RabbitTemplate rabbitTemplate, ObjectMapper objectMapper) {
+        this.repository = repository;
+        this.rabbitTemplate = rabbitTemplate;
+        this.objectMapper = objectMapper;
+    }
+
+    /**
+     * A message should be in a correct format, for example:
+     * {
+     *      "source": "a queue name"
+     * }
+     */
+    @RabbitListener(queues = "${spring.rabbitmq.queue.getFlightsQueue}")
+    public void receiveMessage(String message) {
+        try {
+            GetFlightsEvent getFlightsEvent = objectMapper.readValue(message, GetFlightsEvent.class);
```

Comment:
You can replace implicit json-to-object mapping by using the `org.springframework.amqp.support.converter.MessageConverter` implementation, for example `org.springframework.amqp.support.converter.Jackson2JsonMessageConverter`

Reply:
And then we will be able to remove the comment above the method.
For the future, in cases when we need to write java docs with some parameter exmlanation, you can use `@param`, for example:
```
/**
*
* @param message massage in format <code>{"source": "a queue name"}</code>>
*/
...
```

Reply:
Thanks for suggestions, I used them ;)